src/services/users.js

```diff
@@ -35,9 +37,25 @@ module.exports = (env, jwt, database, sendEmail, mediaClient, validations) => {
     return jwt.sign({ code: random() }, { expiresIn: '24h' })
   }
 
+  let whiteListedDomains = []
+  try {
+    whiteListedDomains = env('WHITELISTED_DOMAINS').split(',').map(str => str.toLowerCase().trim())
+  } catch (e) {
+    console.error(`WHITELISTED_DOMAINS not set as comma delimited string of email addresses properly, error: ${e.stack}`)
+  }
+
+  // Allow all domains when no domains have been selected or check to see if the domain is one that has been whitelisted
+  const isDomainAuthorized = domain => !whiteListedDomains.length || whiteListedDomains.includes(domain)
+
   return {
     login (email, password, client) {
       email = normalizeEmail(email)
+      if (!isDomainAuthorized(getEmailDomain(email))) {
+        const error = `non-whitelisted user "${email}" attempted to login`
+        console.log(error)
+        return Promise.reject(error)
```

Comment:
We should use the `reject` method from line 25? It has to be used with a string like this: `reject('INVALID_CREDENTIALS')` and then you will find a key in the i18n file for the message.